src/sqlfluff/core/linter.py

```diff
@@ -152,15 +160,80 @@ def get_violations(
             violations = [v for v in violations if not v.ignore]
             # Ignore any rules in the ignore mask
             if self.ignore_mask:
-                for line_no, rules in self.ignore_mask:
```

Comment:
Moved this logic into a new function to simplify testing. The logic itself is more complex now with "enable" and "disable".